zproject/urls.py

```diff
@@ -222,6 +222,10 @@
         {'GET': 'zerver.views.streams.list_subscriptions_backend',
          'POST': 'zerver.views.streams.add_subscriptions_backend',
          'PATCH': 'zerver.views.streams.update_subscriptions_backend'}),
+    # PATCH update stream name
+    url(r'^rename_stream$', 'zerver.lib.rest.rest_dispatch',
+        {'PATCH': 'zerver.views.streams.json_rename_stream'}),
+
```

Comment:
I think a potentially cleaner approach would be to make this just another thing one can do in the PATCH method that leads to `update_stream_backend`.  What do you think about that @baali?